mcnode/ds.go

```diff
@@ -105,6 +108,42 @@ func (ds *RocksDS) Delete(key Key) error {
 	return ds.db.Delete(ds.wo, key[2:])
 }
 
+func (ds *RocksDS) Sync() error {
+	return ds.db.Flush(ds.fo)
+}
+
+func (ds *RocksDS) IterKeys(ctx context.Context) (<-chan Key, error) {
```

Comment:
It's probably ok, but the key=hash invariant is more widespread than just Put/PutBatch; the key type is a typedef for a multihash, and there are several places where the keys are decoded as mhashes.
Having said that, we could in theory extend to a more generalized interface, but I am not sure we'll gain anything from it.
Also, there is the issue of tracking references for arbitrary object, garbage collection would just reap whatever data we put in.